File: src/main/java/p00/Problem_0039.java
package p00;

public class Problem_0039 {

	public static int solve() {

		int highest_amount = 0;
		int p = 0;

		for (int i = 3; i <= 1000; i++) {

			int count = 0;

			for (int c = i / 3 + 1; c < i / 2; c++) {
				int squared = (int) Math.pow(c, 2);

				for (int a = 1; a <= (i - c) / 2; a++) {
					double b = Math.sqrt(squared - Math.pow(a, 2));

					if (a + b + c == i)
						count++;
				}

			}

			if (count > highest_amount) {
				highest_amount = count;
				p = i;
			}

		}
		return p;
	}

}

File: src/main/kotlin/p00/Problem0039.kt
package p00

import kotlin.math.pow
import kotlin.math.sqrt

class Problem0039 {
    fun solve(): Int {
        var highestAmount = 0
        var p = 0
        for (i in 3..1000) {
            var count = 0
            for (c in i / 3 + 1 until i / 2) {
                val squared = c.toDouble().pow(2.0).toInt()
                for (a in 1..(i - c) / 2) {
                    val b = sqrt(squared - a.toDouble().pow(2.0))
                    if (a + b + c == i.toDouble()) count++
                }
            }
            if (count > highestAmount) {
                highestAmount = count
                p = i
            }
        }
        return p
    }
}